src/lib.rs

```diff
@@ -2357,12 +2357,13 @@ impl Build {
                     cmd.args.push("-Wl,-melf_i386".into());
                 }
 
+                if (target.starts_with("arm") || target.starts_with("thumb"))
+                    && target.ends_with("-none-eabihf")
```

Comment:
This would have the effect of changing the behavior of all the linux, android, apple, etc. targets as well, which carries more risk of subtle breakage, IMO. The existing places where this was added that I moved are either implicitly only for -none- targets, or not consistent with thumb vs arm versions of the same hosted target, where this should not be needed. (M-profile and R-profile are only ever -none-.)

Reply:
Thanks for explanation!

Reply:
Update, I wasn't checking the tier 3 targets as they don't show up on `rustup target list`, and there are already targets that have inconsistent flags applied to them based on the current rules. I think it is okay to fix these, based on how GCC already behaves for these targets.